stubs/urllib3/urllib3/exceptions.pyi

```diff
@@ -1,50 +1,87 @@
+from email.errors import MessageDefect
+from http.client import IncompleteRead as httplib_IncompleteRead
 from typing import Any
 
+from urllib3.connectionpool import ConnectionPool, HTTPResponse
+from urllib3.util.retry import Retry
+
 class HTTPError(Exception): ...
 class HTTPWarning(Warning): ...
 
 class PoolError(HTTPError):
-    pool: Any
-    def __init__(self, pool, message) -> None: ...
-    def __reduce__(self): ...
+    def __init__(self, pool: ConnectionPool, message: str) -> None: ...
+    def __reduce__(self) -> tuple[Any, tuple[Any, ...]]: ...
 
 class RequestError(PoolError):
-    url: Any
-    def __init__(self, pool, url, message) -> None: ...
-    def __reduce__(self): ...
+    def __init__(self, pool: ConnectionPool, url: str, message: str) -> None: ...
+    def __reduce__(self) -> tuple[Any, tuple[Any, ...]]: ...
 
 class SSLError(HTTPError): ...
-class ProxyError(HTTPError): ...
+
+class ProxyError(HTTPError):
+    original_error: Exception
+    def __init__(self, message: str, error: Exception, *args: Any) -> None: ...
+
 class DecodeError(HTTPError): ...
 class ProtocolError(HTTPError): ...
 
-ConnectionError: Any
+ConnectionError = ProtocolError
 
 class MaxRetryError(RequestError):
-    reason: Any
-    def __init__(self, pool, url, reason=...) -> None: ...
+    reason: Exception | Any
```

Comment:
Why `Any` and not `None`?

Reply:
So I think my initial thinking was so that you wouldn't need to use `if max_retry_error.reason is not None:` blocks. Sort of like we do for function returns.

That being said, the `reason` arg defaults to `None` so if someone uses `max_retry_error.reason` then it probably is a good idea for them to use `if max_retry_error.reason is not None:`. So will update to `reason: Exception | None` as I think you're right 👍

Reply:
updated now